crates/turborepo-scm/src/package_deps.rs

```diff
@@ -0,0 +1,148 @@
+use std::{collections::HashMap, process::Command};
+
+use turbopath::{AbsoluteSystemPathBuf, AnchoredSystemPathBuf, RelativeUnixPathBuf};
+
+use crate::{hash_object::hash_objects, ls_tree::git_ls_tree, status::append_git_status, Error};
+
+pub type GitHashes = HashMap<RelativeUnixPathBuf, String>;
+
+pub fn get_package_file_hashes_from_git_index(
+    turbo_root: &AbsoluteSystemPathBuf,
+    package_path: &AnchoredSystemPathBuf,
+) -> Result<GitHashes, Error> {
+    // TODO: memoize git root -> turbo root calculation once we aren't crossing ffi
+    let git_root = find_git_root(turbo_root)?;
+    let full_pkg_path = turbo_root.resolve(package_path);
+    let git_to_pkg_path = git_root.anchor(&full_pkg_path)?;
+    let pkg_prefix = git_to_pkg_path.to_unix()?;
+    let mut hashes = git_ls_tree(&full_pkg_path)?;
+    // Note: to_hash is *git repo relative*
+    let to_hash = append_git_status(&full_pkg_path, &pkg_prefix, &mut hashes)?;
+    hash_objects(&full_pkg_path, to_hash, &pkg_prefix, &mut hashes)?;
+    Ok(hashes)
+}
+
+pub(crate) fn find_git_root(
+    turbo_root: &AbsoluteSystemPathBuf,
+) -> Result<AbsoluteSystemPathBuf, Error> {
+    let rev_parse = Command::new("git")
+        .args(["rev-parse", "--show-cdup"])
+        .current_dir(turbo_root)
+        .output()?;
+    let root = String::from_utf8(rev_parse.stdout)?;
+    Ok(turbo_root.join_literal(root.trim_end()).to_realpath()?)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::process::Command;
+
+    use super::*;
+
+    fn tmp_dir() -> (tempfile::TempDir, AbsoluteSystemPathBuf) {
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let dir = AbsoluteSystemPathBuf::new(tmp_dir.path().to_path_buf())
+            .unwrap()
+            .to_realpath()
+            .unwrap();
+        (tmp_dir, dir)
+    }
+
+    fn require_git_cmd(repo_root: &AbsoluteSystemPathBuf, args: &[&str]) {
+        let mut cmd = Command::new("git");
+        cmd.args(args).current_dir(repo_root);
+        assert_eq!(cmd.output().unwrap().status.success(), true);
+    }
+
+    fn setup_repository(repo_root: &AbsoluteSystemPathBuf) {
```

Comment:
Should we reuse the `setup_repository` code from `git.rs`?

Reply:
We can. We'd need to find a natural place for that function to live, right?

Reply:
Yeah we can have a test_utils.rs or something